File: sample/vxlisp/js/test/vx/sample_test.js
'strict mode'

import vx_sample from "../../src/vx/sample.js"
import vx_core from "../../src/vx/core.js"
import vx_test from "../../src/vx/test.js"

export default class vx_sample_test {

  static test_package(context) {
    const testcaselist = vx_sample_test.test_cases(context)
    const output = vx_core.f_new(
      vx_test.t_testpackage,
      ":testpkg", "vx/sample",
      ":caselist", testcaselist,
      ":coveragesummary", vx_sample_test.test_coveragesummary(),
      ":coveragedetail", vx_sample_test.test_coveragedetail()
    );
    return output;
  }

  static test_coveragesummary() {
    return vx_core.f_new(
      vx_test.t_testcoveragesummary,
      "testpkg",   "vx/sample", 
      "constnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 1, ":total", 1), 
      "docnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 50, ":tests", 2, ":total", 4), 
      "funcnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 50, ":tests", 1, ":total", 2), 
      "bigospacenums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 0, ":total", 0), 
      "bigotimenums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 100, ":tests", 0, ":total", 0), 
      "totalnums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 50, ":tests", 2, ":total", 4), 
      "typenums", vx_core.f_new(vx_test.t_testcoveragenums, ":pct", 0, ":tests", 0, ":total", 1)
    )
  }

  static test_coveragedetail() {
    return vx_core.f_new(
      vx_test.t_testcoveragedetail,
      "testpkg", "vx/sample",
      "typemap",
        vx_core.f_new(
          vx_core.t_intmap,
          "mytype", 0
        ),
      "constmap",
        vx_core.f_new(
          vx_core.t_intmap,
          "myconst", 1
        ),
      "funcmap",
        vx_core.f_new(
          vx_core.t_intmap,
          "main", 0,
          "myfunc", 1
        )
    )
  }

  static c_myconst(context) {
    const output = vx_core.f_new(
      vx_test.t_testcase,
      ":passfail", false,
      ":testpkg", "vx/sample",
      ":casename", "myconst",
      ":describelist",
        vx_core.f_new(
          vx_test.t_testdescribelist,
          vx_core.f_new(
            vx_test.t_testdescribe,
            ":describename", "(test 4 myconst)",
            ":testresult",
            vx_test.f_test(
                4,
                vx_sample.c_myconst,
                context
              )
          )
        )
    )
    return output
  }

  static f_myfunc(context) {
    const output = vx_core.f_new(
      vx_test.t_testcase,
      ":passfail", false,
      ":testpkg", "vx/sample",
      ":casename", "myfunc",
      ":describelist",
        vx_core.f_new(
          vx_test.t_testdescribelist,
          vx_core.f_new(
            vx_test.t_testdescribe,
            ":describename", "(test 5 (myfunc 1))",
            ":testresult",
            vx_test.f_test(
              5,
              vx_sample.f_myfunc(1),
              context
            )
          )
        )
    )
    return output
  }

  static test_cases(context) {
    const output = vx_core.f_new(
      vx_test.t_testcaselist,
      vx_sample_test.c_myconst(),
      vx_sample_test.f_myfunc(context)
    )
    return output
  }

}

File: sample/java/src/test/java/com/vxlisp/vx/SampleTest.java

package com.vxlisp.vx;

import java.util.Arrays;
import java.util.ArrayList;
import java.util.List;


public final class SampleTest {

  static Test.Type_testcase c_myconst(final Core.Type_context context) {
    Test.Type_testcase output = Test.t_testcase.vx_new(
      ":passfail", false,
      ":testpkg", "vx/sample",
      ":casename", "myconst",
      ":describelist",
      Test.t_testdescribelist.vx_new(
        Test.t_testdescribe.vx_new(
          ":describename", "(test 4 myconst)",
          ":testresult",
            Test.f_test(
              Core.t_int.vx_new_from_int(4),
              Sample.c_myconst,
              context
            )
        )
      )
    );
    return output;
  }

  static Test.Type_testcase f_myfunc(final Core.Type_context context) {
    Test.Type_testcase output = Test.t_testcase.vx_new(
      ":passfail", false,
      ":testpkg", "vx/sample",
      ":casename", "myfunc",
      ":describelist",
      Test.t_testdescribelist.vx_new(
        Test.t_testdescribe.vx_new(
          ":describename", "(test 5 (myfunc 1))",
          ":testresult",
            Test.f_test(
              Core.t_int.vx_new_from_int(5),
              Sample.f_myfunc(Core.t_int.vx_new_from_int(1)),
              context
            )
        )
      )
    );
    return output;
  }

  public static Test.Type_testcaselist test_cases(final Core.Type_context context) {
    List<Core.Type_any> arraylisttestcase = new ArrayList<>(Arrays.asList(
      SampleTest.c_myconst(context),
      SampleTest.f_myfunc(context)
    ));
    Test.Type_testcaselist output = Test.t_testcaselist.vx_new(arraylisttestcase);
    return output;
  }

  public static Test.Type_testcoveragesummary test_coveragesummary() {
    return Test.t_testcoveragesummary.vx_new(
      ":testpkg",   "vx/sample", 
      ":constnums", Test.t_testcoveragenums.vx_new(":pct", 100, ":tests", 1, ":total", 1), 
      ":docnums", Test.t_testcoveragenums.vx_new(":pct", 50, ":tests", 2, ":total", 4), 
      ":funcnums", Test.t_testcoveragenums.vx_new(":pct", 50, ":tests", 1, ":total", 2), 
      ":ospacenums", Test.t_testcoveragenums.vx_new(":pct", 0, ":tests", 0, ":total", 2), 
      ":otimenums", Test.t_testcoveragenums.vx_new(":pct", 0, ":tests", 0, ":total", 2), 
      ":totalnums", Test.t_testcoveragenums.vx_new(":pct", 50, ":tests", 2, ":total", 4), 
      ":typenums", Test.t_testcoveragenums.vx_new(":pct", 0, ":tests", 0, ":total", 1)
    );
  }

  public static Test.Type_testcoveragedetail test_coveragedetail() {
    return Test.t_testcoveragedetail.vx_new(
      ":testpkg", "vx/sample",
      ":typemap", Core.t_intmap.vx_new(
        ":mytype", 0
      ),
      ":constmap", Core.t_intmap.vx_new(
        ":myconst", 1
      ),
      ":funcmap", Core.t_intmap.vx_new(
        ":main", 0,
        ":myfunc", 1
      )
    );
  }

  public static Test.Type_testpackage test_package(final Core.Type_context context) {
    Test.Type_testcaselist testcaselist = test_cases(context);
    Test.Type_testpackage output = Test.t_testpackage.vx_new(
      ":testpkg", "vx/sample", 
      ":caselist", testcaselist,
      ":coveragesummary", test_coveragesummary(),
      ":coveragedetail", test_coveragedetail()
    );
    return output;
  }

}
